src/data/real/liouville.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import data.real.irrational
+
+/-!
+# Liouville's theorem
+This file will contain a proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+
+At the moment, it contains the definition of a Liouville number and a proof that Liouville
+numbers are irrational.
+
+The commented imports below will appear as they are needed.  I (Damiano) leave them here for
+ease of reference.
+--import data.polynomial.denoms_clearable
+--import ring_theory.algebraic
+--import topology.algebra.polynomial
+--import analysis.calculus.mean_value
+-/
+
+section irrational
+
+/--
+A Liouville number `x` is a number such that for every natural number `n`, there exists `a, b ∈ ℤ`
+with `1 < b` such that `0 < |x - a/b| < 1/bⁿ`.  In the implementation, the condition `x ≠ a/b`
+replaces the traditional equivalent `0 < |x - a/b|`.
+-/
+def is_liouville (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
+  1 < b ∧ x ≠ a / b ∧ abs (x - a / b) < 1 / b ^ n
+
+lemma irrational_of_is_liouville {x : ℝ} (h : is_liouville x) : irrational x :=
```

Comment:
If you call the lemma `is_liouville.irrational`, then it will be possible to use it through dot notation.